hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusQueueProcessors.java

```diff
@@ -0,0 +1,180 @@
+import org.hawkular.cmdgw.command.ws.server.WebSocketHelper;
+
+@Startup
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+public class BusQueueProcessors {
+
+    private static class FeedQueueListener<T extends BasicMessage>
+            extends org.hawkular.bus.common.consumer.AbstractBasicMessageListener<T> {
+        private final ConnectedSessions connectedFeeds;
+
+        public FeedQueueListener(ConnectedSessions connectedFeeds) {
+            super();
+            this.connectedFeeds = connectedFeeds;
+        }
+
+        @Override
+        public void onMessage(Message message) {
+            try {
+                String feedId = message.getStringProperty(Constants.HEADER_FEEDID);
+                String basicMessageClass = message.getStringProperty(Constants.HEADER_BASIC_MESSAGE_CLASS);
+                Class<T> cl = (Class<T>) Class.forName(basicMessageClass);
+                if (feedId != null) {
+                    Session session = connectedFeeds.getSession(feedId);
+                    BasicMessageWithExtraData<T> request = parseMessage(message, cl);
+                    new WebSocketHelper().sendSync(session, request);
+                } else {
+                    // TODO send a message back to the UI saying that there is no such feed
+                }
+            } catch (Exception e) {
+                MsgLogger.LOG.errorCannotProcessExecuteOperationResponseMessage(e);
+            }
+        }
+
+    }
+
+    private static class UiQueueListener<T extends BasicMessage>
+            extends org.hawkular.bus.common.consumer.AbstractBasicMessageListener<T> {
+        private final CommandContextFactory commandContextFactory;
+        private final Commands commands;
+
+        public UiQueueListener(CommandContextFactory commandContextFactory, Commands commands) {
+            super();
+            this.commandContextFactory = commandContextFactory;
+            this.commands = commands;
+        }
+
+        @Override
+        public void onMessage(Message message) {
+            try {
+                String basicMessageClass = message.getStringProperty(Constants.HEADER_BASIC_MESSAGE_CLASS);
+                Class<T> cl = (Class<T>) Class.forName(basicMessageClass);
+                BasicMessageWithExtraData<T> messageWithData = parseMessage(message, cl);
+                T request = messageWithData.getBasicMessage();
+                Command<BasicMessage> command = (Command<BasicMessage>) commands.getCommand(request.getClass());
+                CommandContext context = commandContextFactory.newCommandContext(null);
+                command.execute(request, messageWithData.getBinaryData(), context);
+            } catch (Exception e) {
+                MsgLogger.LOG.errorCannotProcessExecuteOperationResponseMessage(e);
+            }
+        }
+
+        protected void onBasicMessage(BasicMessageWithExtraData<T> messageWithData) {
+        }
```

Comment:
As I said above the assumption of one Listener implementation per Message type does not hold here and therefore I cannot use the deserialization in the abstract superclass. The question is if the present concept of storing the `basicMessageClassName` in the bus message header could/should be moved to the parent listener in the bus to replace the type param based resolution there?